backend/src/main/java/gov/cdc/usds/simplereport/service/AddressValidationService.java

```diff
@@ -0,0 +1,74 @@
+package gov.cdc.usds.simplereport.service;
+
+import com.smartystreets.api.ClientBuilder;
+import com.smartystreets.api.exceptions.SmartyException;
+import com.smartystreets.api.us_street.Candidate;
+import com.smartystreets.api.us_street.Client;
+import com.smartystreets.api.us_street.Lookup;
+import com.smartystreets.api.us_street.MatchType;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlArgumentException;
+import gov.cdc.usds.simplereport.config.simplereport.SmartyStreetsConfig;
+import gov.cdc.usds.simplereport.db.model.auxiliary.StreetAddress;
+import gov.cdc.usds.simplereport.service.errors.InvalidAddressException;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AddressValidationService {
+
+  private Client _client;
+
+  public AddressValidationService(Client client) {
+    _client = client;
+  }
+
+  @Autowired
+  public AddressValidationService(SmartyStreetsConfig config) {
+    _client =
+        new ClientBuilder(config.getSmartyAuthId(), config.getSmartyAuthToken())
+            .buildUsStreetApiClient();
+  }
+
+  public Lookup createLookup(
+      String street1, String street2, String city, String state, String postalCode) {
+    Lookup lookup = new Lookup();
+    lookup.setStreet(street1);
+    // Smartystreets defines Street2 as "Any extra address information (e.g., Leave it on the front
+    // porch.)"
+    // and secondary as "Apartment, suite, or office number (e.g., "Apt 52" or simply "52"; not
+    // "Apt52".)""
+    lookup.setSecondary(street2);
+    lookup.setCity(city);
+    lookup.setState(state);
+    lookup.setZipCode(postalCode);
+    return lookup;
+  }
+
+  /** Returns a StreetAddress if the address is valid and throws an exception if it is not */
+  public StreetAddress getValidatedAddress(Lookup lookup) throws InvalidAddressException {
+    lookup.setMatch(MatchType.STRICT);
+    try {
+      _client.send(lookup);
+    } catch (SmartyException | IOException ex) {
+      throw new IllegalGraphqlArgumentException(
+          "The server is unable to verify the address you entered. Please try again later");
+    }
+
+    ArrayList<Candidate> results = lookup.getResult();
+
+    if (results.isEmpty()) {
```

Comment:
does the Lookup tell us anything else about the address validation issue when it fails?

Reply:
Sadly it does not. The SmartyStreets API response for an invalid address is an empty array

Reply:
I wonder if failing is the right thing to do then? What if a region isn't covered by SmartyStreets (like a tribal reservation)? Seems like this exception would prevent overriding it without editing the database?

Reply:
Support for non USPS shippable housing is a feature on the roadmap. I think we are still in the design/ research phase but it will probably be the addition of a check box to enter a more free form loaction input instead of an address.

The philosophy here is that it is much easier to loosen up our requirements in the future if there is an issue then it is to lock them down.

Reply:
i'm not 100% clear on what the outcome of our spec conversation was yesterday, but it seems we just need to make sure we're unilaterally blocking people with unrecognizable addresses

Reply:
My understanding is for now we will are not be enforcing any verification on the person address. With the plan of using the frontend to strongly suggest the user enter a known USPS shippable address.

We will be keeping the enforcing a valid USPS shippable address on the Facility and ordering provider